lib/spack/spack/cmd/installer/scripts/spack.bat

```diff
@@ -1,2 +1,211 @@
+::
+:: 'spack load' is smarter than just 'load' because it converts its
+:: arguments into a unique Spack spec that is then passed to module
+:: commands.  This allows the user to use packages without knowing all
+:: their installation details.
+::
+:: e.g., rather than requiring a full spec for libelf, the user can type:
+::
+::     spack load libelf
+::
+:: This will first find the available libelf module file and use a
+:: matching one.  If there are two versions of libelf, the user would
+:: need to be more specific, e.g.:
+::
+::     spack load libelf@0.8.13
+::
+:: This is very similar to how regular spack commands work and it
+:: avoids the need to come up with a user-friendly naming scheme for
+:: spack module files.
+::#######################################################################
+
+:_sp_shell_wrapper
+set "_sp_flags="
+set "_sp_args="
+set "_sp_subcommand="
+setlocal enabledelayedexpansion
+for %%x in (%*) do (
+    set "t=%%x"
+    if "!t:~0,1!" == "-" (
+        if not defined _sp_flags (
+            set "_sp_flags=!t!"
+            shift
+        ) else (
+            set "_sp_flags=!_sp_flags! !t!"
+            shift
+            )
+        ) else (
+            if not defined _sp_subcommand (
+                set "_sp_subcommand=!t!"
+                shift
+            ) else (
+                set "_sp_args=!_sp_args! !t!"
+            )
+        )
+    )
+:: h and V flags don't require further output parsing.
+:: Boolean operators do not exist for non integers in batch
+if defined _sp_flags (
+    if NOT "%_sp_flags%"=="%_sp_flags:-h=%" (
+        if not defined _sp_subcommand (
+            python %spack% %_sp_flags% %_sp_subcommand% %_sp_args%
+            exit /B 0
+        ) else (
+            python %spack% %_sp_subcommand% %_sp_flags% %_sp_args%
+            exit /B 0 )
+    )
+    if NOT "%_sp_flags%"=="%_sp_flags:-V=%" (
+        python %spack% %_sp_flags% %_sp_subcommand% %_sp_args%
+        exit /B 0
+    )
+)
+:: pass parsed variables outside of local scope. Need to do
+:: this because delayedexpansion can only be set by setlocal
+echo %_sp_flags%>flags
+echo %_sp_args%>args
+echo %_sp_subcommand%>subcmd
+endlocal
+set /p _sp_subcommand=<subcmd
+set /p _sp_flags=<flags
+set /p _sp_args=<args
+if "%_sp_subcommand%"=="ECHO is off." (set "_sp_subcommand=")
+if "%_sp_flags%"=="ECHO is off." (set "_sp_flags=")
+if "%_sp_args%"=="ECHO is off." (set "_sp_args=")
+del subcmd
+del flags
+del args
+
+:: Filter out environment commands. For any others, just run the command.
+if "%_sp_subcommand%" == "cd" (
+    goto :case_cd
+) else if "%_sp_subcommand%" == "env" (
+    goto :case_env
+) else if "%_sp_subcommand%" == "load" (
+    goto :case_load
+) else if "%_sp_subcommand%" == "unload" (
+    goto :case_load
+) else (
+    python "%spack%" %_sp_subcommand% %_sp_flags% %_sp_args%
+    goto :end_switch
+)
+::#######################################################################
+
+:case_cd
+if "%_sp_flags%" == "" (
```

Comment:
Should this whole block be removed? I think it's intended to match setup-env but it's never called.